addons/mail/models/mail_channel.py

```diff
@@ -769,8 +768,11 @@ def channel_info(self):
             # find the channel partner state, if logged user
             if self.env.user and self.env.user.partner_id:
                 info['message_needaction_counter'] = channel.message_needaction_counter
-                info['message_unread_counter'] = channel.message_unread_counter
                 partner_channel = member_of_current_user_by_channel.get(channel, self.env['mail.channel.partner'])
+                prefetch = []
+                for member in partner_channel:
+                    prefetch.append(member.id)
+
```

Comment:
This should be something like this (not tested)
```suggestion
                partner_channel = member_of_current_user_by_channel.get(channel, self.env['mail.channel.partner']).with_prefetch([m.id for m in member_of_current_user_by_channel.values()])
```